tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -170,7 +171,22 @@ def get_logs(
         except Exception as ex:
             log.warning("Failed to retrieve logs. %s", ex, exc_info=True)
 
-        super().get_logs(device.get_id(), date_from=date_from, show=show)
+        log_output = super().get_logs(device.get_id(), date_from=date_from, show=False)
+        if show:
+            hide_sensitive = self._hide_sensitive_factory()
+            for line in log_output:
+                print(hide_sensitive(line))
+
+    def _hide_sensitive_factory(self):
+        # This is fragile and should be improved upon once a more suitable/robust method of logging and querying
+        # the mqtt messages is found.
```

Comment:
Maybe a bit fragile, but in any case not too harmful as keeping a witness of the redacted log entry.